mithril-stm/src/schnorr_signature/signature.rs

```diff
@@ -0,0 +1,116 @@
+use anyhow::{Result, anyhow};
+use midnight_circuits::hash::poseidon::PoseidonChip;
+use midnight_circuits::instructions::HashToCurveCPU;
+use midnight_circuits::instructions::hash::HashCPU;
+use midnight_curves::{Fq as JubjubBase, Fr as JubjubScalar, JubjubSubgroup};
+
+use group::Group;
+
+use crate::{
+    Index,
+    schnorr_signature::{
+        DST_LOTTERY, DST_SIGNATURE, JubjubHashToCurve, SchnorrVerificationKey, get_coordinates,
+        hash_msg_to_jubjubbase, jubjub_base_to_scalar,
+    },
+};
+
+/// Structure of the Schnorr signature to use with the SNARK
+/// This signature includes a value `sigma` which depends only on
+/// the message and the signing key.
+/// This value is used in the lottery process to determine the correct indices.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub(crate) struct SchnorrSignature {
```

Comment:
All the fields should have a doc comment in all the structs for better readability and documentation.

+1 on the non descriptive names mentioned by @curiecrypt .